tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java

```diff
@@ -479,4 +480,20 @@ private static Module chooseDropListsModule(DistributedJamesConfiguration config
 
         };
     }
+
+    private static Module chooseDeletedMessageVault(VaultConfiguration vaultConfiguration) {
+        if (vaultConfiguration.isEnabled() && vaultConfiguration.isWorkQueueEnabled()) {
+            return Modules.combine(
+                new DistributedDeletedMessageVaultModule(),
+                new DeletedMessageVaultRoutesModule());
+        }
+        if (vaultConfiguration.isEnabled()) {
+            return Modules.combine(
+                new CassandraDeletedMessageVaultModule(),
+                new DeletedMessageVaultRoutesModule());
+        }
```

Comment:
How about JMAP routes for the vault: does it makes sense to bind them when vault is disabled?

Reply:
It makes it crash, just fixed.